packages/cozy-harvest-lib/src/helpers/oauth.js

```diff
@@ -41,15 +45,26 @@ export const checkOAuthData = (konnector, data) => {
  * ```
  * @return {boolean} `true` if an OAuth response has been handled, `false` otherwise
  */
-export const handleOAuthResponse = () => {
+export const handleOAuthResponse = (options = {}) => {
+  let realtime = options.realtime
+  if (!realtime) {
+    let client = options.client
+    if (!client) {
+      const root = document.querySelector('[role=application]')
+      client = new CozyClient({
+        uri: `${window.location.protocol}//${root.dataset.cozyDomain}`,
+        token: root.dataset.cozyToken
+      })
```

Comment:
We should not need to create a new cozy-client here. Isn't there already created at app level ? It could be passed to this function.

Reply:
For example, in the home application, this function is called before any initialization ( https://github.com/cozy/cozy-home/blob/d89e562dacb1cbbe7f743488d6972f990fc24e7f/src/targets/browser/index.jsx#L17 ) to intercept OAuth redirect uri.

The other solution would be that the host application sends the cozy-client or cozy-realtime instance itself (which is possible with the new handleOAuthResponse signature).

But I don't want this change to cause regressions in the different applications using cozy-harvest-lib.

Reply:
Je vois [ici la méthode loadInstanceOptionsFromDOM](https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/CozyClient.js#L116), à mon avis ce serait bien de l'utiliser. il me semble par contre qu'elle ne va pas appeler login() même si on arrive à avoir l'URL du cozy depuis le DOM, ca pourrait être qqchose à corriger. On a déjà des méthodes statiques fromEnv, fromOldClient, ca aurait sûrement du sens de faire la même chose. Est ce que tu peux voir si tu peux rajouter une méthode fromDOM dans cozy-client et potentiellement l'utiliser ici ?

Reply:
Since this is not really part of this PR and since this PR already provides value, if this is urgent, I'd be OK to merge now if we fix the problem later.